### DISCUSSION TITLE
flow documentation

### DISCUSSION THREAD

**Discussion by shihabkb** [openintegrationhub/openintegrationhub#1162] [Q&A] [answered]
Can I get some documentation about flows?

Some of my questions are,

1.Like what is nodes? 
2. How the nodes are executing? 
3. What is edge? 
4. What is source and target in edge?

**Comment by RobinBrinkmann** [comment, marked as answer]
Hi @shihabkb and thanks for your question. The best way to start is probably the [installation guide](https://openintegrationhub.github.io/docs/3%20-%20Deployment/LocalInstallationGuide.html). Judging from your questions though, this may be where you are coming from already. So I will try to give you some additional explanation here.

Generally a flow describes the data exchange between several systems. Let's say you want to get incoming orders from a webshop and send them on to an accounting solution. Then your nodes are a description of these steps: first step (node) = getOrders from webshop. Second step (node) = createOrder in accounting solution. The edges describe the direction / sequence. You want to get data from the webshop (your source) and put it into the accounting solution (target) and not the other way around. This of course becomes more meaningful when your flow has more than just two steps. 

Flows get executed by several services in the framework. Mainly the component orchestrator and the ferryman. Basically, once a flow is activated, data flows between the defined steps. This can either be triggered by webhooks or a polling mechanism. Also see [here](https://openintegrationhub.github.io/docs/1%20-%20BasicConcepts/FlowExecution.html).

I hope this helped. If you have more technical questions, feel free to ask or join our [slack channel](https://join.slack.com/t/openintegrationhub/shared_invite/zt-kwbj0osy-hgzhKJJmYn4tXeVN42udtg)

**Reply by RobinBrinkmann** [reply]
Please also see our latest docs update: https://openintegrationhub.github.io/docs/1%20-%20BasicConcepts/FlowBasics.html

**Comment by shihabkb** [comment]
Thanks Robin.
